src/Worker/Core/Shims/TaskOrchestrationShim.cs

```diff
@@ -30,9 +31,24 @@ partial class TaskOrchestrationShim : TaskOrchestration
     public TaskOrchestrationShim(
         OrchestrationInvocationContext invocationContext,
         ITaskOrchestrator implementation)
+        : this(invocationContext, implementation, new Dictionary<string, object?>())
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TaskOrchestrationShim"/> class.
+    /// </summary>
+    /// <param name="invocationContext">The invocation context for this orchestration.</param>
+    /// <param name="implementation">The orchestration's implementation.</param>
+    /// <param name="properties">Configuration for the orchestration.</param>
+    public TaskOrchestrationShim(
+        OrchestrationInvocationContext invocationContext,
+        ITaskOrchestrator implementation,
+        Dictionary<string, object?> properties)
```

Comment:
Can this be read-only?
```suggestion
        IReadOnlyDictionary<string, object?> properties)
```